pwnlib/tubes/ssh.py

```diff
@@ -667,8 +667,12 @@ def process(self, argv=None, executable=None, tty=True, cwd=None, env=None, time
                 See ``stdin``.
             preexec_fn(callable):
                 Function which is executed on the remote side before execve().
+                This **MUST** be a self-contained function -- it must perform
+                all of its own imports, and cannot refer to variables outside
+                its scope.
```

Comment:
I thought you chould just check if `func_closure` is `None/()` and if `func_code.co_names` only contains names from `__builtins__`.

However `def f(): import x` also has `x` in its `co_names`. So I guess you are right.